### PR TITLE
Adding wchar_t to char adaptation

### PR DESCRIPTION
**Pull request by MitraDarja**
Fix for #231 by adding wchar_t.

### CODE REVIEW

**Comment by h-2** [comment]
@eseiler Can you review this tomorrow?

File: include/seqan3/alphabet/adaptation/char.hpp [hunk 1]
```diff
@@ -69,10 +69,12 @@ namespace seqan3::detail
 //!\brief The list of types that are defined as char adaptations.
 using char_adaptations            = meta::list<char,
                                                char16_t,
-                                               char32_t>;
+                                               char32_t,
+                                               wchar_t>; // TODO Windows adaption requires different handling of wchar_t because in Windows it is 16 bits and holds UTF-16 code units
```
Comment (eseiler) [inline include/seqan3/alphabet/adaptation/char.hpp]: This line is longer than 120 characters.

(See https://github.com/seqan/seqan3/wiki/General-formatting )

File: include/seqan3/alphabet/adaptation/char.hpp [hunk 2]
```diff
@@ -98,7 +100,7 @@ namespace seqan3
 // ------------------------------------------------------------------
 
 /*!\brief Specialisation of seqan3::underlying_char for char types.
- * \tparam char_type One of `char`, `char16_t` or `char32_t`.
+ * \tparam char_type One of `char`, `char16_t` , `char32_t` or `wchar_t`.
```
Comment (eseiler) [inline include/seqan3/alphabet/adaptation/char.hpp]: Please remove the whitespace before the comma.

(Same for the other 8 changes)

**Comment by MitraDarja** [comment]
Fixed comments in #298

**Pull request merged**